sql/src/main/java/io/crate/operation/projectors/ShardingShardRequestAccumulator.java

```diff
@@ -165,6 +194,32 @@ private ShardId getShardId(String indexName, String id, @Nullable String routing
         }
     }
 
+    private static class ShardLocation {
+
+        private final ShardId shardId;
+        private final String nodeId;
+
+        public ShardLocation(ShardId shardId, String nodeId) {
+            this.shardId = shardId;
+            this.nodeId = nodeId;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            ShardLocation that = (ShardLocation) o;
+
+            return shardId.equals(that.shardId);
```

Comment:
Because we are using this as a key in requestsByShard. I don't think we should change the semantics of this (given we might have null and/or relocating nodeId)